pkg/controllers/webhook/controller.go

```diff
@@ -1,218 +1,859 @@
+	if err != nil {
 		return err
 	}
-	if err := c.reconcileValidatingWebhookConfiguration(ctx, logger, name); err != nil {
-		return err
+	for _, policy := range policies {
+		policyKey, err := cache.MetaNamespaceKeyFunc(policy)
+		if err != nil {
+			return err
+		}
+		ready := true
+		for _, set := range c.policyState {
+			if !set.Has(policyKey) {
+				ready = false
+			}
+		}
+		if policy.IsReady() != ready {
+			policy = policy.CreateDeepCopy()
+			status := policy.GetStatus()
+			status.SetReady(ready)
+			if policy.GetNamespace() == "" {
+				_, err := c.kyvernoClient.KyvernoV1().ClusterPolicies().UpdateStatus(ctx, policy.(*kyvernov1.ClusterPolicy), metav1.UpdateOptions{})
+				if err != nil {
+					return err
+				}
+			} else {
+				_, err := c.kyvernoClient.KyvernoV1().Policies(policy.GetNamespace()).UpdateStatus(ctx, policy.(*kyvernov1.Policy), metav1.UpdateOptions{})
+				if err != nil {
+					return err
+				}
+			}
+		}
 	}
 	return nil
 }
+
+func (c *controller) reconcile(ctx context.Context, logger logr.Logger, key, namespace, name string) error {
+	switch name {
+	case config.MutatingWebhookConfigurationName:
+		if c.runtime.IsRollingUpdate() {
+			c.enqueueResourceWebhooks(1 * time.Second)
+		} else {
+			if err := c.reconcileResourceMutatingWebhookConfiguration(ctx); err != nil {
+				return err
+			}
+			if err := c.updatePolicyStatuses(ctx); err != nil {
+				return err
+			}
+		}
+	case config.ValidatingWebhookConfigurationName:
+		if c.runtime.IsRollingUpdate() {
+			c.enqueueResourceWebhooks(1 * time.Second)
+		} else {
+			if err := c.reconcileResourceValidatingWebhookConfiguration(ctx); err != nil {
+				return err
+			}
+			if err := c.updatePolicyStatuses(ctx); err != nil {
+				return err
+			}
+		}
+	case config.PolicyValidatingWebhookConfigurationName:
+		return c.reconcilePolicyValidatingWebhookConfiguration(ctx)
+	case config.PolicyMutatingWebhookConfigurationName:
+		return c.reconcilePolicyMutatingWebhookConfiguration(ctx)
+	case config.VerifyMutatingWebhookConfigurationName:
+		return c.reconcileVerifyMutatingWebhookConfiguration(ctx)
+	}
+	return nil
+}
+
+func (c *controller) buildVerifyMutatingWebhookConfiguration(caBundle []byte) (*admissionregistrationv1.MutatingWebhookConfiguration, error) {
+	return &admissionregistrationv1.MutatingWebhookConfiguration{
+			ObjectMeta: objectMeta(config.VerifyMutatingWebhookConfigurationName),
+			Webhooks: []admissionregistrationv1.MutatingWebhook{{
+				Name:         config.VerifyMutatingWebhookName,
+				ClientConfig: c.clientConfig(caBundle, config.VerifyMutatingWebhookServicePath),
+				Rules: []admissionregistrationv1.RuleWithOperations{{
+					Rule: verifyRule,
+					Operations: []admissionregistrationv1.OperationType{
+						admissionregistrationv1.Update,
+					},
+				}},
+				FailurePolicy:           &ignore,
+				SideEffects:             &noneOnDryRun,
+				ReinvocationPolicy:      &ifNeeded,
+				AdmissionReviewVersions: []string{"v1beta1"},
```

Comment:
Should we think about updating the `AdmissionReviewVersion` to `v1` in coming release ?

Reply:
we should consider changing that but 1.9 seems a more reasonable target IMHO.

Reply:
Yup make sense in 1.9